### PR TITLE
Add the capability to log any metaData together with the message.

### PR DESCRIPTION
**Pull request by sli-cs**
Add the capability to log any metaData together with the message.
modify README.md a bit

### CODE REVIEW

**Comment by sli-cs** [comment]
Updated the PR with some refactoring.

File: README.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ cokeLogger.debug("this is debug line %d", 10);
 
 cokeLogger.info('test message %s, %s', 'first', 'second');
 
+//logging the message together with metaData as the 1st argument
+var metaInfo = {"provider": "ISP", "appID": 234};
+cokeLogger.info(metaInfo, 'The message will be logged together with some metaInfo');
```
Comment (vr000m) [inline README.md:43]: so the metaInfo can be any JSON object?

File: cs-logger.js [hunk 2]
```diff
@@ -29,11 +29,34 @@ var _level = {
...
 
+/* we consider the first argument in the log(xx...) as a metaData if it is an Object (not Array or String or number)*/
+var _objectConstructor = {}.constructor;
+function isMetaData(anObj) {
+    if (anObj.constructor === _objectConstructor) {
+        return true;
+    }
+    return false;
+}
+
```
Comment (vr000m) [inline cs-logger.js:42]: @marcinag: do you have comments about this check?

File: cs-logger.js [hunk 3]
```diff
@@ -29,11 +29,34 @@ var _level = {
...
+// copied from http://stackoverflow.com/questions/171251/how-can-i-merge-properties-of-two-javascript-objects-dynamically
+/**
+ * Overwrites obj1's values with obj2's and adds obj2's if non existent in obj1
+ * @param obj1
+ * @param obj2
+ * @returns obj3 a new object based on obj1 and obj2
+ */
+function merge_options(obj1,obj2){
```
Comment (vr000m) [inline cs-logger.js]: merge_objects instead?

**Comment by sli-cs** [comment]
Merging this to master after addressing all the comments.

**Pull request merged**